core/src/avm1/globals/sound.rs

```diff
@@ -38,10 +38,12 @@ pub fn constructor<'gc>(
 ) -> Result<Value<'gc>, Error<'gc>> {
     // 1st parameter is the movie clip that "owns" all sounds started by this object.
     // `Sound.setTransform`, `Sound.stop`, etc. will affect all sounds owned by this clip.
-    let owner = args
-        .get(0)
-        .map(|o| o.coerce_to_object(activation))
-        .and_then(|o| o.as_display_object());
+    let owner = if !args.is_empty() {
+        let start_clip = activation.target_clip_or_root()?;
+        activation.resolve_target_display_object(start_clip, *args.get(0).unwrap(), false)?
```

Comment:
Looks good, but let's avoid the `.unwrap()`. The best way I know is:
```rs
let owner = if let [owner, ..] = args {
    let start_clip = activation.target_clip_or_root()?;
    activation.resolve_target_display_object(start_clip, *owner, false)?
} else {
    None
};
```

Reply:
Thank you, that's good to know. 😉